prerelease/action.yaml

```diff
@@ -5,67 +5,198 @@ inputs:
     required: false
     description: Perform checkout as first step of action
     default: "true"
+  checkout-fetch-depth:
+    required: false
+    description: The number of commits to fetch. 0 indicates all history for all branches and tags
+    default: "0"
+  create-prerelease:
+    required: false
+    description: Whether semantic-release should create a prerelease or do a dry run. This can be useful to set to true when a prerelease requires pushing artifacts semantic-release is in charge of generating
+    default: "false"
   github-token:
     description: GitHub token that can checkout the consumer repository as well as create tags/releases against it. e.g. 'secrets.GITHUB_TOKEN'
     required: true
-  go-version:
-    description: Go version to use for building
-    required: true
-    default: 1.17.3
-  push-docker-snapshot:
-    description: If a docker snapshot image is generated, push it to the to the registry
-    required: false
-    default: "false"
   docker-username:
     description: Docker username to push the snapshot image to the registry
     required: false
   docker-password:
     description: Docker password to push the snapshot image to the registry
     required: false
+  extra-plugins:
+    required: false
+    description: Extra plugins for pre-install. You can also specify specifying version range for the extra plugins if you prefer.  Defaults to install @open-turo/semantic-release-config.
+    default: |
+      @open-turo/semantic-release-config
 outputs:
   version:
     description: Version of the project
     value: ${{ steps.release.outputs.version }}
 runs:
   using: composite
   steps:
+    - name: Set vars
```

Comment:
Here there is a lot of logic that we are duplicating from `actions-node` and `actions-java`. I'm sure there is a better way to not DRY these things